### PR TITLE
Make evp_proxy configurable via env vars

### PR DESCRIPTION
**Pull request by albertvaka**
### What does this PR do?

Calls `BindEnv` for the new settings we added in https://github.com/DataDog/datadog-agent/pull/12271

Also renames files form `evp_intake_proxy` to `evp_proxy` since that's the name we settled on.

### Motivation

I forgot this on my first PR.

### CODE REVIEW

File: pkg/config/evp_proxy.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package config
+
+// setupEvpProxy initializes the configuration values of the evp_proxy.
+func setupEvpProxy(cfg Config) {
+	cfg.BindEnv("evp_proxy_config.enabled")
```
Comment (gbbr) [inline pkg/config/evp_proxy.go:10]: Any reason you wanted a separate file and function? I'd just put them in `pkg/config/config.go`

Comment (albertvaka) [reply pkg/config/evp_proxy.go:10]: I saw that appsec got a separate file so I did the same, but I agree it can be on the same one. Let me change it.

**Review by gbbr — APPROVED** [review]
Thanks Albert 🙇🏻 I think the AppSec change was the anomaly 🙂

**Review by Kaderinho — APPROVED** [review]
LGTM for Agent Core files

**Pull request merged**